tb/cocotb/soc/test_boot.py

```
@@ -248,47 +258,86 @@ async def test_boot_sram_sentinels(dut):
-        f"expected SW0 = 0x{SW0_INSN:08x}"
+    # Full PC-trace assert: every committed PC must match the golden model exactly.
+    assert collected_pcs == _GOLDEN_PCS, (
+        f"PC trace mismatch (got {len(collected_pcs)} commits, "
+        f"expected {len(_GOLDEN_PCS)}):\n"
+        + "\n".join(
+            f"  [{i}] got=0x{got:08x} exp=0x{exp:08x}"
+            + (" <-- MISMATCH" if got != exp else "")
+            for i, (got, exp) in enumerate(zip(collected_pcs, _GOLDEN_PCS))
+        )
+        + (
+            f"\n  extra commits: "
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)